osu.Game/Overlays/MusicController.cs

```diff
@@ -290,15 +290,8 @@ private void changeBeatmap(WorkingBeatmap newWorking)
 
             current = newWorking;
 
-            if (!audioEquals || CurrentTrack.IsDummyDevice)
```

Comment:
Not sure I wholly agree with the `IsDummyDevice` check being removed due to "not making sense at face value". Dummy/virtual tracks are cheap to recreate, and so it makes *some* sense to not transfer virtual tracks to avoid lifetime/shared reference bugs.

Reply:
You mention "lifetime/shared reference bugs", but the hope is that we never want to let that happen in the first place unless it makes sense (i.e. `WaveformTestBeatmap`/`ClockBackedTestWorkingBeatmap` storing their track store locally), instead of just adding an early-guard blindly like that and potentially hiding greater issues at hand. Unsure.

Reply:
If we want to be as conservative about transferring tracks around as possible, then it makes sense to never transfer tracks which are cheap to construct and don't need the optimisation, no?

Reply:
I mean, the reason why we have never seen this issue of potentially transferring disposed tracks on `ClockBackedTestWorkingBeatmap` is because of this specific condition, until the issue became visible with `WaveformTestBeatmap` which returns non-virtual tracks.

But coming at it from a non-testing perspective, I suppose it makes sense that we don't have to transfer virtual tracks given that they are cheap and don't need the "optimisation" this method provides.